hub/util/namedtuple.py

```diff
@@ -0,0 +1,87 @@
+from typing import Iterable, List
+from collections import OrderedDict
+
+
+def namedtuple(T: str, fields: List[str]):
+    """
+    Similar to `collections.namedtuple`, but uses subscripting instead of
+    dot operator to access fields to allow arbitrary string field names.
+
+    Example:
+
+    T = namedtuple("T", ["a", "b", "c"])
+    t = T(1, 2, 3)
+    # or
+    t = T(a=1, b=2, c=3)
+
+    assert t['a'] == 1
+    assert t['b'] == 2
+    assert t['c'] == 3
+
+    for x in t:
+        print(x)
+
+    # >>> 1
+    # >>> 2
+    # >>> 3
+    """
+
+    class NamedTuple(object):
+        # For pytorch dataloader compatibility
+        __class__ = tuple  # type: ignore
+        _fields = fields
+
+        def __init__(self, *args, **kwargs):
```

Comment:
can we move this class outside of a method body and rename it? Something like `SubscriptNamedTuple`

Reply:
also, if you could rename the filename and test filename to represent this that would be cool

Reply:
The class gets the fields arg from from the context of the method.